ansible/roles/vm_set/library/vm_topology.py

```diff
@@ -426,10 +445,21 @@ def remove_dut_if_from_docker(self, iface_name, dut_iface):
 
         return
 
-    def add_veth_if_to_docker(self, ext_if, int_if):
+    def add_veth_if_to_docker(self, ext_if, int_if, create_vlan_subintf=False, **kwargs):
+        """Create vethernet devices (ext_if, int_if) and put int_if into the ptf docker."""
+        if create_vlan_subintf:
+            try:
+                vlan_subintf_sep = kwargs["sub_interface_separator"]
+                vlan_subintf_vlan_id = kwargs["sub_interface_vlan_id"]
+            except KeyError:
+                raise TypeError("Missing arguments for function 'add_veth_if_to_docker'")
+
         self.update()
 
-        t_int_if = hashlib.md5((PTF_NAME_TEMPLATE % self.vm_set_name).encode("utf-8")).hexdigest()[0:6] + int_if + '_t'
+        t_int_if = hashlib.md5((PTF_NAME_TEMPLATE % self.vm_set_name).encode("utf-8")).hexdigest()[0:6]
+        if create_vlan_subintf:
+            int_sub_if = int_if + vlan_subintf_sep + vlan_subintf_vlan_id
+            t_int_sub_if = t_int_if + vlan_subintf_sep + vlan_subintf_vlan_id
 
         if t_int_if in self.host_ifaces:
```

Comment:
The format of t_int_if is different because of the changes in L459

Reply:
This is to avoid the interface name length limitation, added a function to do this.